pkg/bootkube/create.go

```diff
@@ -43,143 +59,133 @@ func CreateAssets(config clientcmd.ClientConfig, manifestDir string, timeout tim
 		return true, nil
 	}
 
-	createFn := func() (bool, error) {
-		err := createAssets(config, manifestDir)
-		if err != nil {
-			err = fmt.Errorf("Error creating assets: %v", err)
-			glog.Error(err)
-			UserOutput("%v\n", err)
-			UserOutput("\nNOTE: Bootkube failed to create some cluster assets. It is important that manifest errors are resolved and resubmitted to the apiserver.\n")
-			UserOutput("For example, after resolving issues: kubectl create -f <failed-manifest>\n\n")
-		}
-
-		// Do not fail cluster creation due to missing assets as it is a recoverable situation
-		// See https://github.com/kubernetes-incubator/bootkube/pull/368/files#r105509074
-		return true, nil
-	}
-
 	UserOutput("Waiting for api-server...\n")
-	start := time.Now()
 	if err := wait.Poll(5*time.Second, timeout, upFn); err != nil {
 		err = fmt.Errorf("API Server is not ready: %v", err)
 		glog.Error(err)
 		return err
 	}
 
 	UserOutput("Creating self-hosted assets...\n")
-	timeout = timeout - time.Since(start)
-	if err := wait.PollImmediate(5*time.Second, timeout, createFn); err != nil {
-		err = fmt.Errorf("Failed to create assets: %v", err)
-		glog.Error(err)
-		return err
+	if err := creater.createManifests(m); err != nil {
+		UserOutput("%v\n", err)
+		UserOutput("\nNOTE: Bootkube failed to create some cluster assets. It is important that manifest errors are resolved and resubmitted to the apiserver.\n")
+		UserOutput("For example, after resolving issues: kubectl create -f <failed-manifest>\n\n")
+		return fmt.Errorf("failed to create assets: %v", err)
```

Comment:
We previously had allowed bootkube to continue even if creation events failed. This was primarily because failing early actually made it more difficult (often times) to debug the issue (usually unrelated to actually bringing up a cluster). So if bootkube continued until completion (and left you with an accessible apiserver) you could then debug the failures through kubectl. If you just exit early - you might have nothing to look into.

This is also somewhat related to comment below about retrying the creation with a PUT (at very least shouldn't exit on already exist errors).

Reply:
:+1: will modify to not error if creating the resource fails and avoid the PUT.